music21/expressions.py

```diff
@@ -572,19 +758,21 @@ def realize(self, srcObj: 'music21.note.Note', *, inPlace=False):
             use_ql = srcObj.duration.quarterLength / 4
 
         remainderQL = srcObj.duration.quarterLength - (2 * use_ql)
-        if self.direction == 'down':
-            transposeInterval = self.size.reverse()
-        else:
-            transposeInterval = self.size
+        transposeInterval = self.getSize(srcObj.pitch)
         mordNotes: list[note.Note] = []
         self.fillListOfRealizedNotes(srcObj, mordNotes, transposeInterval, useQL=use_ql)
 
         currentKeySig = srcObj.getContextByClass(key.KeySignature)
         if currentKeySig is None:
             currentKeySig = key.KeySignature(0)
 
-        for n in mordNotes:
-            n.pitch.accidental = currentKeySig.accidentalByStep(n.step)
+        # second (middle) note might need an accidental from the keysig (but
+        # only if it doesn't already have an accidental from accid)
+        for noteIdx, n in enumerate(mordNotes):
+            noteNum: int = noteIdx + 1
```

Comment:
no need for typing here -- noteNum will be typed properly from `enumerate`

Reply:
Huh.  I don't seem capable of anything between "no typing" and "always typing unless I really can't".  Unless you have a simple rule I can apply to find a comfortable spot in between those two places?

Reply:
I mean, it's fine to have it here, just redundant.  Generally you'll want typing on all inputs and outputs of functions, so the internal things that need typing are attributes (since they'll be queried/set and need to know what's allowed there) and especially variables that have containers, like `notes: list[note.Note] = []` 

Are you using an IDE that infers type when you mouse over a variable?  If the IDE can't infer the type, then it needs to be typed.  If a variable is a return value from a typed function (like enumerate) then it doesn't need a type (like nodeIdx), and `int + int` magically types an int to still be an int.